### PR TITLE
Port walrus on windows

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

**Comment by clover2123** [comment]
Please rebase this patch too

**Comment by ksh8281** [comment]
Rebase done!

File: build/target.cmake [hunk 1]
```diff
@@ -1,14 +1,121 @@
...
+    SET (WALRUS_CXXFLAGS_STATICLIB -fPIC -DWALRUS_EXPORT=)
+    SET (WALRUS_CXXFLAGS_SHELL -DWALRUS_EXPORT= -frtti -std=c++17)
+ELSE()
+    MESSAGE (FATAL_ERROR ${CMAKE_CXX_COMPILER_ID} " is Unsupported Compiler")
+ENDIF()
+
+# Default options per host
 IF (${WALRUS_HOST} STREQUAL "linux")
+    FIND_PACKAGE (PkgConfig REQUIRED)
+    IF (WALRUS_THREADING)
```
Comment (clover2123) [inline build/target.cmake:116]: What is this `WALRUS_THREADING` option for?
It seems that this option is nowhere used now

Comment (ksh8281) [reply build/target.cmake]: I removed that option

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**